### PR TITLE
GUI: initial log viewer

### PR DESCRIPTION
**Pull request by Megamouse**
Adds a rudimentary log viewer with color highlighting.
I'll add a search tool later.
Let's see if this builds at all first.

### CODE REVIEW

**Comment by Megamouse** [comment]
Added search tool

**Comment by Megamouse** [comment]
I decided to merge this as there isn't really anything wrong with it.
It's just kinda slow and unpolished.

But I lost interest for now and maybe someone else is gonna use and improve it

**Comment by Nekotekina** [comment]
Maybe fix new prefix for "always" logs?

**Comment by Megamouse** [comment]
I don't know what to fix..
You merged it btw ;)

File: rpcs3/rpcs3qt/syntax_highlighter.cpp [hunk 1]
```diff
@@ -52,6 +53,18 @@ void Highlighter::highlightBlock(const QString &text)
 	}
 }
 
+LogHighlighter::LogHighlighter(QTextDocument* parent) : Highlighter(parent)
+{
+	//addRule("^[^·].*$", gui::utils::get_label_color("log_level_always")); // unused for now
```
Comment (Nekotekina) [inline rpcs3/rpcs3qt/syntax_highlighter.cpp:58]: I meant this, you added a regular prefix for "always" logs.

Comment (Megamouse) [reply rpcs3/rpcs3qt/syntax_highlighter.cpp:58]: but the prefix does not appear in the log files, only in the GUI.
So this regex is not gonna apply here

Comment (Nekotekina) [reply rpcs3/rpcs3qt/syntax_highlighter.cpp:58]: Ah nvm then. My bad.

**Pull request merged**